src/vendor/vendor.c

```diff
@@ -1834,39 +1809,46 @@ int DiscoverDM_ForAllComponents(char *objs_filename, char *params_filename)
 ** \return  USP_ERR_OK if successful
 **
 **************************************************************************/
-int DiscoverDM_ForComponent(rdk_component_t *rdkc, kv_vector_t *rdk_objects, kv_vector_t *rdk_params)
+int DiscoverDM_ForComponent(kv_vector_t *rdk_objects, kv_vector_t *rdk_params)
```

Comment:
The name of this function and the header comment needs to change. It isn't discovering the data model for a single component anymore, but is discovering the whole data model.